### DISCUSSION TITLE
How to implement the IMetadataEndpointAddressProvider

### DISCUSSION THREAD

**Discussion by loferreiranuno** [CoreWCF/CoreWCF#853] [Q&A]
We are trying to develope a WCF service inside kubernetes, using a reverse proxy via https.
Our service is http.
Our services are generating an invalid wsdl ( because the endpoint of the wsdl is http and reverse proxy is expecting https ). 
 
We have an implementation for the IMetadataEndpointAddressProvider, and we use the MetadataEndpointAddressServiceBehavior. 

But the our wsdl now is missing the "location" value that indicates to the next wsdl. 
`<wsdl:import namespace="http://xxx.xxx" location=""/>`

If we use the singleWsdl all works fine. But the wsdl , it doesn't.

Is there any other way to fix that? 

UseRequestHeadersForMetadataAddressBehavior creates correct the location value, but the endpoints are not correct.

**Comment by szyb** [comment]
Having the same issue, but slightly different (probably due to newer version of CoreWCF). The schemaLocation attribute is not filled in within `xsd:import` node.
```<xsd:import schemaLocation="" namespace="http://schemas.datacontract.org/2004/07/[datacontract namespace here]"/>```
Using singleWsdl works fine (thanks for tip @loferreiranuno!), however it is still not clear why wsdl is not well formed.
The same issue in inner XSD definitions (i.e. with parameter ?xsd=xsd0)

**Comment by vait** [comment]
Hey! I try fix this problem in https://github.com/CoreWCF/CoreWCF/pull/1101
